account-kit/react/src/hooks/useSolanaConnection.ts

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import * as solanaNetwork from "../solanaNetwork.js";
+import { SolanaSigner } from "@account-kit/signer";
+import { useSolanaSigner } from "./useSolanaSigner.js";
+import { getSolanaConnection, watchSolanaConnection } from "@account-kit/core";
+import { useSyncExternalStore } from "react";
+import { useAlchemyAccountContext } from "./useAlchemyAccountContext.js";
+
+/**
+ * Returned from the solana connection.
+ */
+export interface SolanaConnection {
+  /** The solana signer used to send the transaction */
+  readonly signer: SolanaSigner | null;
+  /** The solana connection used to send the transaction */
+  readonly connection: solanaNetwork.Connection | null;
+}
+
+/**
+ * The parameters for the SolanaConnectionHookParams hook.
+ */
+export type SolanaConnectionHookParams = {
+  signer?: SolanaSigner;
+};
+
+/**
+ * This is the hook that will be used to send to the solana connection.
```

Comment:
nit: but I think this jsdoc description is a little confusing.

Maybe could be something like "This hook is used for establishing a connection to Solana and returns the connection object and the signer object."